### PR TITLE
Option to not update configs with AWS tokens

### PR DESCRIPTION
**Pull request by mchristie**
Although it's recommended to use AWS roles that's not always possible.

If the application is setting AWS keys and secrets with credentials which do not need an AWS token, adding the token in from the function role will cause the credentials to become invalid.

This PR adds a config option which disables updating the AWS configs with AWS_SESSION_TOKEN

### CODE REVIEW

File: config/bref.php [hunk 1]
```diff
@@ -30,4 +30,17 @@
...
+    | Fix AWS role credentials
+    |--------------------------------------------------------------------------
+    |
+    | Laravel does not use AWS_SESSION_TOKEN environment vars by default
+    | Bref can automatically add these tokens into your configuration
+    | If you are not using AWS Roles you may need to disable this.
+    |
+    */
+
+    'aws_session_tokens' => true,
```
Comment (tillkruss) [inline config/bref.php:44]: Let's call this `use_session_token`.

File: src/BrefServiceProvider.php [hunk 2]
```diff
@@ -39,16 +39,15 @@ public function register()
         $this->fixDefaultConfiguration();
 
         Config::set('app.mix_url', Config::get('app.asset_url'));
-
+        
         Config::set('trustedproxy.proxies', ['0.0.0.0/0', '2000:0:0:0:0:0:0:0/3']);
-
+
```
Comment (tillkruss) [inline src/BrefServiceProvider.php:44]: Remove extra whitespace.

File: src/BrefServiceProvider.php [hunk 3]
```diff
@@ -39,16 +39,15 @@ public function register()
...
         Config::set('view.compiled', StorageDirectories::Path . '/framework/views');
         Config::set('cache.stores.file.path', StorageDirectories::Path . '/framework/cache');
-
-        Config::set('cache.stores.dynamodb.token', env('AWS_SESSION_TOKEN'));
-        Config::set('filesystems.disks.s3.token', env('AWS_SESSION_TOKEN'));
-        Config::set('queue.connections.sqs.token', env('AWS_SESSION_TOKEN'));
-        Config::set('services.ses.token', env('AWS_SESSION_TOKEN'));
+        
+        if (Config::get('bref.aws_session_tokens', true)) {
+            $this->fixAWSCredentials();
```
Comment (tillkruss) [inline src/BrefServiceProvider.php:49]: Let's just inline the code, no need to extract it into a method.

**Review by tillkruss — CHANGES_REQUESTED** [review]

File: config/bref.php [hunk 4]
```diff
@@ -41,6 +41,6 @@
     |
     */
 
-    'aws_session_tokens' => true,
+    'use_session_tokens' => true,
```
Comment (tillkruss) [inline config/bref.php:44]: @mnapoli: I'd say "session token", what do you reckon?

Comment (mnapoli) [reply config/bref.php:44]: +1 for the singular version since it's one token

File: config/bref.php [hunk 5]
```diff
@@ -30,4 +30,17 @@
...
     'request_context' => false,
 
+    /*
+    |--------------------------------------------------------------------------
+    | Use AWS credential tokens
+    |--------------------------------------------------------------------------
+    |
+    | Laravel does not use AWS_SESSION_TOKEN environment vars by default
+    | Bref can automatically add these tokens into your configuration
+    | If you are not using AWS Roles you may need to disable this.
```
Comment (mnapoli) [inline config/bref.php:40]: Uploading AWS credentials in Lambda is a very last resort option honestly, I'd love to know when it's not possible to use the Lambda credentials?

Anyway, I'd optimize for being super explicit that most users should not do this/use this, unless they definitely know why they are doing this. So I'd reword probably like this:

```suggestion
    | Bref fixes Laravel to use the AWS_SESSION_TOKEN environment variable
    | provided by AWS Lambda automatically (required for AWS credentials to work).
    | You should disable that only if you set your own AWS access keys manually
    | (which is not recommended in most cases).
```

Comment (mchristie) [reply config/bref.php:40]: Totally agree it's a last resort. Our AWS setup has a lot of legacy and we have many clients with DKIM etc in SES in one region/account, but for various reasons we're also sending emails from another AWS account. We use roles for most things, but in this case we need to use different credentials for connecting to SES.

**Review by mnapoli — APPROVED** [review]

**Comment by deleugpn** [comment]
My take on this is that the behavior of the adapter is to casually adjust Laravel's default installation for a quick drop-in replacement for Lambda. If you're deploying your project into Lambda it's an opportunity to replace secret keys with IAM Roles.
However, if you have reasons to stick with access key / secret key, all you'd have to do is change the name of the driver / connection you're using. E.g. if instead of using the `s3` disk on your filesystems config, you could just name it something else like `bucket` or `storage` or the business meaning behind the files that get stored there. Anything that is not the default laravel installation configuration wouldn't be touched by the Bridge and you'd be free to configure it however you feel like it.
It seems like a low effort on the user's part and one less configuration that Bref would have to maintain.

**Review by georgeboot — APPROVED** [review]

**Comment by mnapoli** [comment]
Yeah, I'd tend to agree with @deleugpn as this is a niche use case. I'm not against merging this, but I'm not especially pushing for it. I'll let you guys decide.
